### ISSUE TITLE
An option to disable all the `Pick`s in favor of duplicated type definitions

### ISSUE THREAD

**Issue by MarkKahn** [dotansimha/graphql-code-generator#2020]
This may already exist in some form, I've just stated investigating using your library since I love the idea of having customizable templates for our codebase.

**Is your feature request related to a problem? Please describe.**

This is extremely hard (impossible since not all the info is there) to parse.  This is what VS code (and ts cli) shows me as a type (with field names & types redacted, all of the `...`s are literally shown to me):

```tsx
const fieldA: Maybe<{
    __typename: "TypeA";
} & {
    __typename: "TypeA";
} & {
    fieldB: Maybe<{
        __typename: "TypeB";
    } & Pick<TypeB, "c" | "d" | "e" | "f"> & {
        fieldG: {
            __typename: "TypeC";
        } & Pick<...>;
        fieldH: Maybe<...>;
        fieldI: Maybe<...>;
    }>;
} & {
    ...;
} & {
    ...;
} & {
    ...;
} & Pick<...>>
```

**Describe the solution you'd like**

The Apollo client:codegen collapses all of these, and the definitions it generates for this exact case are:

```tsx
export interface TypeA {
  __typename?: 'TypeA';
  fieldA: TypeB | null;
  fieldB: TypeC | null;
  fieldC: TypeD | null;
  fieldD: string | null;
  fieldE: string | null;
  fieldF: string | null;
  ... more fields
}
```

while this makes the types larger it makes the *significantly* easier to parse, and since this is all generated code redundancy really isn't an issue.

**Describe alternatives you've considered**

**Additional context**

**Comment by dotansimha** [comment]
We are thinking about putting the final evaluated primitive type instead of the `Pick<>`, this looks very similar. 
Thank you for opening this issue :)

**Comment by dotansimha** [comment]
@MarkKahn https://github.com/dotansimha/graphql-code-generator/pull/2107 just got merged, and it will be available in the next release.

**Comment by dotansimha** [comment]
Available in 1.4.0 🎉
